src/index/metadata.rs

```diff
@@ -0,0 +1,44 @@
+use crate::package::model::{DependencyReq, PackageVersion};
+use crate::CONFIG;
+
+#[derive(Serialize, Deserialize)]
+pub struct Metadata {
```

Comment:
It seems like we could remove this file and just re-use the facilities provided by the `elba` library instead

Reply:
Yeah, I'll do it.

Reply:
After attempting to replace metadata with elba's IndexEntry, I've found that I have to adpot all elba's name related facilities for that, but it is blocked by some issues that needs to be solved on elba's side, so I prefer to postpond these small problems now and I think we would finnally solve them before 0.3.0 release.

Reply:
If you could open issues on the elba repo for that, that'd be great!